src/content-script/openai.js

```diff
@@ -5,8 +5,29 @@ export async function callGPT4(messages) {
     const storageData = await chrome.storage.local.get(PROFILE_MEMORY_KEY);
     const profileData = storageData?.[PROFILE_MEMORY_KEY] || "";
     const content =
-      "You are a helpful assistant responding to LinkedIn messages on behalf of a the person receiving a certain LinkedIn message." +
-      `Answer any messages concisely, politely, and in a friendly tone. Use this profile data to help formulate your answers: ${profileData}`;
+        "You are a helpful assistant responding to a LinkedIn message from a recruiter on behalf of a the person (candidate) receiving a certain LinkedIn message. " +
+        "Answer any messages concisely, politely, and in a friendly tone. " +
+        "IMPORTANT: You must ALWAYS respond in this exact JSON format:\n" +
+        "{\n" +
+        '  "keyword": "SEND_RESPONSE",  // Use one of: SEND_RESPONSE, REVIEW_DRAFTED_RESPONSE, ASK_PRIVATE_CANDIDATE_INFO, ASK_TO_UPDATE_CANDIDATE_PROFILE\n' +
+        '  "content": "Your actual message here"\n' +
+        "}\n\n" +
+        "Follow this logic when picking the keyword:" +
+        " - SEND_RESPONSE if the information you have available in the profile_data has ALL you need to confidently answer the message without using any other data or assumption. " +
+        "   If you choose this keyword, then the 'content' should be the message to send out. NEVER make assumptions of any kind or use information that is not in the profile_data when crafting your responses.\n" +
+        " - REVIEW_DRAFTED_RESPONSE if the information you have available in the profile_data has probably all you need to answer the message without using any other data or assumption " +
+        "   but you're not sure so the user should still review the message before sending it out. If you choose this keyword, then the 'content' should be the message to review and send out.\n" +
+        " - ASK_PRIVATE_CANDIDATE_INFO if the information you have available in the profile_data ALONE does not have all you need to confidently answer the message." +
```

Comment:
Nice!
I'm not very familiar with AI prompts since working on FE. Can you recommend something to read about that? Or good guide.

Reply:
This guide from OpenAI that has many good points: https://help.openai.com/en/articles/6654000-best-practices-for-prompt-engineering-with-the-openai-api